src/cephadm/cephadm.py

```diff
@@ -4692,6 +4693,35 @@ def command_list_images(ctx: CephadmContext) -> None:
     # print default images
     cp_obj.write(sys.stdout)
 
+
+def update_service_for_daemon(ctx: CephadmContext, daemon_list: list) -> None:
+    """ Update the unit.meta file of daemon with required service name for valid daemons"""
+
+    data = {'service_name': ctx.service_name}
+    update_daemon = ctx.name.split(',')
```

Comment:
I feel like this bit should be done in the `command_update_osd_service` function and then the list of names (what `update_daemon` ends up as) should be passed to this function. As it is now this relies on ctx.name following a specific format which isn't clear when reading this function itself (and also makes it more difficult to ever re-use this if we wanted to)